core/message_parsers.py

```diff
@@ -0,0 +1,350 @@
+                
+            if line.startswith('--') and line.endswith('--'):
+                if current_role:
+                    self.roles.append(GuildRole(current_role, current_members))
+                    current_members = []
+                current_role = line.strip('- ')
+                continue
+                
+            if '●' in line:
+                # Split by bullet points and process each member
+                members = line.split('●')
+                for member in members:
+                    if member.strip():
+                        current_members.append(self._extract_member_info(member))
+                        
+            if line.startswith('Total Members:'):
+                self.total_members = int(re.search(r'\d+', line).group())
+            elif line.startswith('Online Members:'):
+                self.online_members = int(re.search(r'\d+', line).group())
+
+        # Add the last role if exists
+        if current_role:
+            self.roles.append(GuildRole(current_role, current_members))
+
+        return self._format_list_embed()
+
+    def _parse_online_message(self) -> str:
+        self._parse_list_message()  # Reuse list parsing logic
+        # Extract offline members
+        for line in self.raw_message.split('\n'):
+            if line.startswith('Offline Members:'):
+                self.offline_members = int(re.search(r'\d+', line).group())
+                break
+        return self._format_online_embed()
+
+    def _parse_top_message(self) -> str:
+        lines = self.raw_message.split('\n')
+        
+        date_pattern = r'\b(\d{2}/\d{2}/\d{4})\b'
+        match = re.search(date_pattern, self.raw_message)
+        if match:
+            date_str = match.group(1)
+            date_obj = datetime.strptime(date_str, '%m/%d/%Y')
+            self.date = date_obj.date()
+        else:
+            self.date = datetime.now().date()
+        # Parse top entries
+        for line in lines[1:]:  # Skip header
+            if not line.strip():
+                continue
+                
+            match = re.match(r'(\d+)\.\s+(.+?)\s+(\d+,?\d*)\s+Guild Experience', line)
+            if match:
+                position = int(match.group(1))
+                member_text = match.group(2)
+                experience = int(match.group(3).replace(',', ''))
+                
+                member = self._extract_member_info(member_text)
+                member.experience = experience
+                self.top_entries.append(TopEntry(member, experience, position))
+
+        return self._format_top_embed()
+
+    def _format_rank(self, rank: str) -> str:
+        return f"**[{rank}]**" if rank else ""
+
+    def _format_list_embed(self) -> List[discord.Embed]:
+        embeds = []
+        current_description = ""
+        page_number = 1
+
+        current_description += f"# {self.guild_name}\n\n"
+
+        for role in self.roles:
+            role_description = f"## {role.name}\n"
```

Comment:
Can we use the embed title & fields instead?

Reply:
The fields mess up the formatting. I am using Markdown in the description

Reply:
How do they mess up the formatting?